src/parser/mod.rs

```diff
@@ -1271,10 +1271,22 @@ impl<'a> Parser<'a> {
                             )
                         }
                     };
-                    Ok(Expr::CompositeAccess {
-                        expr: Box::new(expr),
-                        key,
-                    })
+                    if self.consume_token(&Token::LParen) {
+                        self.prev_token();
+                        let func = match self.parse_function(ObjectName(vec![key]))? {
+                            Expr::Function(func) => func,
+                            _ => unreachable!(),
```

Comment:
Oh I think this panic won't be desirable here in any case given we don't have an invariant and `self.parse_function` could be in its own right to return something other than `Expr::Function` in a different situation.

Also given the nature of the syntax that we don't have any special keyword to match on, feels a bit uneasy that we could end up parsing sql incorrectly in other dialects in ambiguous cases.

could we do similar to e.g. [try_parse_lambda](https://github.com/apache/datafusion-sqlparser-rs/blob/main/src/parser/mod.rs#L1332-L1346) where this is both specific to mssql dialect, as well as potentially relying on `self.maybe_parse` to bail out if the lookahead isn't what we expect?